Scheduling plugin not updating responding attendee status

miaulalala opened the pull request:
Take 2 on fixing the attendee's own status not updating.

Thanks to @kesselb we now know the correct way to create an iTip message.

From his comment on the other PR: https://github.com/nextcloud/server/pull/28094

The purpose of this pull request is to fix an issue with updating the attendance status for the attendee itself. To reproduce create a event and invite Alice and Bob. Wait for Bob's invitation email and accept the invite. The attendance status for the organizer and Alice is updated (Bob accepted the invite). In Bob's calendar event the attendance status is still pending. 

A possible explanation for this case is the way we handle event invitations. If a user accept or decline an invitation a iTip message is generated. 

https://github.com/nextcloud/server/blob/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/apps/dav/lib/Controller/InvitationResponseController.php#L200-L201

The recipient is used to select the right event from the calendar. When Alice accept the invite an iTip message is generated with sender = alice@example.org and recipient = organizer@example.org. The event for recipient organizer@example.org is selected and the attendance status for alice@example.org updated. Then for every other attendee another iTip message is generated to also update their event. 

https://github.com/nextcloud/3rdparty/blob/6876f1fce8d1c70790c165dd7ed0b4214364e397/sabre/dav/lib/CalDAV/Schedule/Plugin.php#L527-L538

![image](https://user-images.githubusercontent.com/3902676/134508461-e4ce4ea7-d4ee-48d1-b801-2996a32effaa.png)

Sabre assume that we already updated Alice event and there is no need to sent a iTip message to Alice. 

```
$iTipMessage->sender = $row['attendee']; 
$iTipMessage->recipient = $row['attendee']; 
```

When sender = recipient the calendar event for the attendee is updated and a iTip message for the organizer and other attendees generated and properly updated. 

Fixes https://github.com/nextcloud/calendar/issues/2861

miaulalala committed:
Scheduling plugin not updating responding attendee status take two

Signed-off-by: Anna Larch <anna@nextcloud.com>

miaulalala commented:
/backport to stable22

miaulalala commented:
/backport to stable21

miaulalala commented:
/backport to stable20

ChristophWurst commented:
Could you please elaborate on how this fixes the problem? Like what was wrong before and why does this fix the issue. Thanks :)

kesselb committed:
Update attendence for external users

For local users it's possible to select their calendar via the principal url and first update their own attendance status.
External users have no calendar event hence the recipient is the organizer.

Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

st3iny reviewed (approved):
I just tested it on my local instance and it worked great. 

I tested using a second local user and an external email address. The status in my calendar and in the shared event of the second user were updated properly.
